tests/framework/extensions/nodes/ec2/node_status.go

```diff
@@ -0,0 +1,51 @@
+package ec2
+
+import (
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// IsNodeReady is a helper method that will loop and check if the node is ready in the RKE1 cluster.
+// It will return an error if the node is not ready after set amount of time.
+func IsNodeReady(client *rancher.Client, ClusterID string) error {
+	err := wait.Poll(500*time.Millisecond, 30*time.Minute, func() (bool, error) {
+		nodes, err := client.Management.Node.List(&types.ListOpts{
+			Filters: map[string]interface{}{
+				"clusterId": ClusterID,
+			},
+		})
+		if err != nil {
+			return false, err
+		}
+
+		const active = "active"
+
+		for _, node := range nodes.Data {
+			node, err := client.Management.Node.ByID(node.ID)
+			if err != nil {
+				return false, err
+			}
+
+			if node.State == active {
+				logrus.Infof("All nodes in the cluster are in an active state!")
+				return true, nil
+			}
+
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
```

Comment:
It seems like this isn't necessary since all the error returns already have error checks, which error does this check?

Reply:
Less code the better 😄 . I'll remove that as well.